qiita_db/support_files/patches/python_patches/23.py

```diff
@@ -0,0 +1,195 @@
+# schema changes) are done in a single transaction block
+
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import SampleTemplate, PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+# Relaxing constraints on the Sample Template
+# Retrieve all study ids that have a sample template
+sql = """SELECT DISTINCT study_id from qiita.required_sample_info"""
+study_ids = set(s[0] for s in conn_handler.execute_fetchall(sql))
+
+queue_name = "PATCH_23"
+conn_handler.create_queue(queue_name)
+
+sql_select = """SELECT sample_id, physical_location, has_physical_specimen,
+                       has_extracted_data, sample_type, collection_timestamp,
+                       host_subject_id, description, latitude, longitude,
+                       required_sample_info_status
+                FROM qiita.required_sample_info
+                JOIN qiita.required_sample_info_status
+                    USING (required_sample_info_status_id)
+                WHERE study_id=%s"""
+
+sql_cols = """INSERT INTO qiita.study_sample_columns
+                (study_id, column_name, column_type)
+              VALUES (%s, 'physical_specimen_location', 'varchar'),
+                     (%s, 'physical_specimen_remaining', 'bool'),
+                     (%s, 'dna_extracted', 'bool'),
+                     (%s, 'sample_type', 'varchar'),
+                     (%s, 'collection_timestamp', 'timestamp'),
+                     (%s, 'host_subject_id', 'varchar'),
+                     (%s, 'description', 'varchar'),
+                     (%s, 'latitude', 'float8'),
+                     (%s, 'longitude', 'float8'),
+                     (%s, 'required_sample_info_status', 'varchar')"""
+
+sql_alter = """
+ALTER TABLE qiita.{0} ADD COLUMN physical_specimen_location varchar;
+ALTER TABLE qiita.{0} ADD COLUMN physical_specimen_remaining bool;
+ALTER TABLE qiita.{0} ADD COLUMN dna_extracted bool;
+ALTER TABLE qiita.{0} ADD COLUMN sample_type varchar;
+ALTER TABLE qiita.{0} ADD COLUMN collection_timestamp timestamp;
+ALTER TABLE qiita.{0} ADD COLUMN host_subject_id varchar;
+ALTER TABLE qiita.{0} ADD COLUMN description varchar;
+ALTER TABLE qiita.{0} ADD COLUMN latitude float8;
+ALTER TABLE qiita.{0} ADD COLUMN longitude float8;
+ALTER TABLE qiita.{0} ADD COLUMN required_sample_info_status varchar;"""
+
+sql_update = """UPDATE qiita.{0}
+                SET physical_specimen_location=%s,
+                    physical_specimen_remaining=%s,
+                    dna_extracted=%s,
+                    sample_type=%s,
+                    collection_timestamp=%s,
+                    host_subject_id=%s,
+                    description=%s,
+                    latitude=%s,
+                    longitude=%s,
+                    required_sample_info_status=%s
+                WHERE sample_id=%s"""
+
+for s_id in study_ids:
+    # Get the data from the requried_sample_info
+    data = conn_handler.execute_fetchall(sql_select, (s_id,))
```

Comment:
That would mean queues aren't appropriate here, right? 

The chance of something going wrong is low, but the risk is high, unless there's something I'm misunderstanding?